experimental/dds/tree2/src/test/shared-tree/fuzz/targetedFuzz.spec.ts

```diff
@@ -3,69 +3,80 @@
 import {
-	AsyncGenerator,
-	makeRandom,
-	performFuzzActionsAsync,
-	SaveInfo,
-} from "@fluid-internal/stochastic-test-utils";
+	Client,
+	DDSFuzzHarnessEvents,
+	defaultDDSFuzzSuiteOptions,
+	runTestForSeed,
+	// eslint-disable-next-line import/no-internal-modules
+} from "@fluid-internal/test-dds-utils/dist/ddsFuzzHarness";
+import { TypedEventEmitter } from "@fluidframework/common-utils";
 import { moveToDetachedField, compareUpPaths, rootFieldKeySymbol, UpPath } from "../../../core";
 import { brand } from "../../../util";
-import {
-	TestTreeProvider,
-	SummarizeType,
-	initializeTestTree,
-	validateTree,
-	toJsonableTree,
-} from "../../utils";
-import { FuzzTestState, makeOpGenerator, EditGeneratorOpWeights } from "./fuzzEditGenerators";
+import { toJsonableTree, validateTree } from "../../utils";
+import { SharedTreeFactory } from "../../../shared-tree";
+import { singleTextCursor } from "../../../feature-libraries";
+import { makeOpGenerator, EditGeneratorOpWeights } from "./fuzzEditGenerators";
 import { fuzzReducer } from "./fuzzEditReducers";
-import { initialTreeState, makeTree, runFuzzBatch, testSchema } from "./fuzzUtils";
+import { initialTreeState, runFuzzBatch, testSchema } from "./fuzzUtils";
 import { Operation } from "./operationTypes";
 
 export async function performFuzzActionsAbort(
-	generator: AsyncGenerator<Operation, FuzzTestState>,
+	generator: AsyncGenerator<Operation, DDSFuzzTestState<SharedTreeFactory>>,
 	seed: number,
 	saveInfo?: SaveInfo,
-): Promise<FuzzTestState> {
-	const random = makeRandom(seed);
-	const provider = await TestTreeProvider.create(1, SummarizeType.onDemand);
-	const tree = provider.trees[0];
-
-	initializeTestTree(provider.trees[0], initialTreeState, testSchema);
-	validateTree(provider.trees[0], [initialTreeState]);
-
-	// building the anchor for anchor stability test
-	const cursor = tree.forest.allocateCursor();
-	moveToDetachedField(tree.forest, cursor);
-	cursor.enterNode(0);
-	cursor.getPath();
-	cursor.firstField();
-	cursor.getFieldKey();
-	cursor.enterNode(1);
-	const firstAnchor = cursor.buildAnchor();
-	cursor.free();
-
-	const initialState: FuzzTestState = {
-		random,
-		trees: provider.trees,
-		testTreeProvider: provider,
-		numberOfEdits: 0,
+): Promise<DDSFuzzTestState<SharedTreeFactory>> {
+	const baseModel: DDSFuzzModel<
+		SharedTreeFactory,
+		Operation,
+		DDSFuzzTestState<SharedTreeFactory>
+	> = {
+		workloadName: "SharedTree",
+		factory: new SharedTreeFactory(),
+		generatorFactory: () => generator,
+		reducer: fuzzReducer,
+		validateConsistency: () => {},
+	};
+	let firstAnchor;
+	const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
+	emitter.on("clientCreate", (client: Client<SharedTreeFactory>) => {
+		client.channel.storedSchema.update(testSchema);
+		const field = client.channel.editor.sequenceField({
+			parent: undefined,
+			field: rootFieldKeySymbol,
+		});
+		field.insert(0, singleTextCursor(initialTreeState));
+	});
```

Comment:
IMO this test would be a bit cleaner if you handled this in the SharedTreeFactory rather than using this event. It would also give more natural code sharing between the few tests you have.

Reply:
updated to `SharedTreeTestFactory`, and initialize trees using the factory's `onCreate` emitter